pkg/storage/gcs.go

```diff
@@ -155,8 +170,33 @@ func (s *gcsStorage) Open(ctx context.Context, path string) (ExternalFileReader,
 // function; the second argument is the size in byte of the file determined
 // by path.
 func (s *gcsStorage) WalkDir(ctx context.Context, opt *WalkOption, fn func(string, int64) error) error {
-	// TODO, implement this if needed
-	panic("Unsupported Operation")
+	if opt == nil {
+		opt = &WalkOption{}
+	}
+	prefix := path.Join(s.gcs.Prefix, opt.SubDir)
+	if len(prefix) > 0 && !strings.HasSuffix(prefix, "/") {
+		prefix += "/"
+	}
+
+	query := &storage.Query{Prefix: prefix}
```

Comment:
consider doing https://pkg.go.dev/cloud.google.com/go/storage#Query.SetAttrSelection to the query.

Reply:
done.